src/error.rs

```diff
@@ -86,8 +86,28 @@ pub enum Error {
     Transport(Transport),
 }
 
-// Any error that is not a status code error. For instance, DNS name not found,
-// connection refused, or malformed response.
+impl Error {
+    pub fn into_transport(self) -> Option<Transport> {
```

Comment:
I'm more thinking about these, since it consumes self, it's `into_xxx`: https://doc.rust-lang.org/nightly/std/result/enum.Result.html#method.into_ok

Reply:
If you scroll up a little from there, note the trait bound:

```rust
impl<T, E> Result<T, E>
where
    E: Into<!>, 
```

In other words, this is only available on `Result<T, !>`, something that can never error. That's why it returns `T` instead of `Option<T>` like `.ok()`.

Since we return `Option<T>`, I think `.ok()` is the closer analogy rather than `.into_ok()`.

Reply:
Ok. Point taken. This one then :) https://doc.rust-lang.org/std/io/struct.Error.html#method.into_inner

Reply:
Also do we want corresponding `as_transport` / `as_status`? https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
Alright, I am convinced that `into_` naming that returns an `Option` has a good analogy in the stdlib (and in error handling specifically!). I'm `Ok(())` with this as it stands. :D

Reply:
My motivation for these are to be able to write:

`ureqerror.into_transport().map(|t| t.message())`

I.e. a way to go from enum to `Option` for chaining stuff to option. For non-chaining cases I prefer this one.

```rs
if let Error::Transport(t) = error {
   ...
}
```

Unsure what help the reference style `as_` would be. Maybe they are useful?

Reply:
I think think an `as_transport() -> Option<&Transport>` would work better for the chaining case you're describing, because you don't _have_ to consume the `Error` just to get at the message.